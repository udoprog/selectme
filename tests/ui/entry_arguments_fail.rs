fn main() {}

#[selectme::main]
fn main_with_argument(value: u32) {}

#[selectme::main]
pub(crate) fn non_empty_pub_crate(value: u32) {}
