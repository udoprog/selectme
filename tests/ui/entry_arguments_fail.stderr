error: functions marked with `#[selectme::main]` must be `async`
 --> tests/ui/entry_arguments_fail.rs:4:1
  |
4 | fn main_with_argument(value: u32) {}
  | ^^

error: functions marked with `#[selectme::main]` must not take any arguments
 --> tests/ui/entry_arguments_fail.rs:4:22
  |
4 | fn main_with_argument(value: u32) {}
  |                      ^^^^^^^^^^^^

error: functions marked with `#[selectme::main]` must be `async`
 --> tests/ui/entry_arguments_fail.rs:7:1
  |
7 | pub(crate) fn non_empty_pub_crate(value: u32) {}
  | ^^^

error: functions marked with `#[selectme::main]` must not take any arguments
 --> tests/ui/entry_arguments_fail.rs:7:34
  |
7 | pub(crate) fn non_empty_pub_crate(value: u32) {}
  |                                  ^^^^^^^^^^^^
